src/components/select/select.component.spec.ts

```diff
@@ -387,6 +408,54 @@ describe('Select Component - NgModel Input', () => {
 
     });
 
+    fit('should not open dropdown when tabbing past select', (done) => {
+        fixture.detectChanges();
+
+        function focusInput() {
```

Comment:
`Promise.resolve()` is an alternative to `setTimeout`, it is almost an equivalent of a timeout with a time of 0, only quicker.

So you wouldn't need both. In the case where you need to wait for a period of time, eg: 200ms, there are other approaches to avoid actually having to wait that amount of time.

The first one is `whenStable` which returns a promise that completes when Javascripts task queue is empty: https://angular.io/guide/testing#whenstable

Here is how you could use it in your test for example:
```typescript
it('should open dropdown when entering value', async () => {
        fixture.detectChanges();

        const input = getInput();
        input.focus();
        dispatchKeyboardEvent(input, 'keydown', O);
        component.value = 'O';

        await fixture.whenStable();

        fixture.detectChanges();
        expect(getTypeahead()).toBeTruthy();
    });
```

There are other cases where you want to simulate the passing of time you can use the `fakeAsync` and `tick` testing functions Angular provides: https://angular.io/guide/testing#async-test-with-fakeasync